app/controllers/posts_controller.rb

```diff
@@ -588,6 +588,10 @@ def deleted_posts
     render_serialized(posts, AdminUserActionSerializer)
   end
 
+  def pending
+    render_serialized(current_user.pending_posts.order(created_at: :desc), PendingPostSerializer, root: :pending_posts)
```

Comment:
I', with @ZogStriP on this one, we should have the username param in this route, i.e. `posts/:username/pending` and then pull the user from the params. This allows staff members to debug issues if/when necessary and it would be consistent with similar routes (deleted_posts, flagged_posts).

Reply:
This will add unnecessary complexity both on client side and server side IMO:
- server side we’ll have to get the user from the params, check if they can access pending posts for that user, etc.
- client side it’s more or less the same, we need to check current user can see pending posts through the user route (so yes for admins and owner but no for other users).

If we didn’t have access to pending posts through an other existing UI I probably wouldn’t argue here, just to make my position clearer :sweat_smile: 
But if that’s the way we want to do it then I guess I’ll make the change.

Reply:
Ok done, just need to adapt some tests and write some new ones.